[BUG] Parallel merging of same documents leads to error

### Description

This is a very niche scenario, but while playing around with the merge feature, I noticed an error when having two merge-processes running on the same documents.

When you start two merge-processes on the same documents, the second one fails because it cannot find the file anymore, since both use the same tmp filename(?)

As far as I understand, this could also happen when two users share these documents and both users decide to merge them at the same time (not a small timeframe for huge documents). 

### Steps to reproduce

Start two merge processes on the same documents in parallel

### Webserver logs

```bash
[2024-06-26 10:17:21,002] [ERROR] [paperless.consumer] The following error occurred while storing document 2350_2351_merged.pdf after parsing: [Errno 2] No such file or directory: '/tmp/paperless/2350_2351_merged.pdf'
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 327, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/consumer.py", line 698, in run
    self.input_doc.original_file.unlink()
  File "/usr/local/lib/python3.11/pathlib.py", line 1147, in unlink
    os.unlink(self)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/paperless/2350_2351_merged.pdf'
[2024-06-26 10:17:21,010] [ERROR] [paperless.tasks] ConsumeTaskPlugin failed: 2350_2351_merged.pdf: The following error occurred while storing document 2350_2351_merged.pdf after parsing: [Errno 2] No such file or directory: '/tmp/paperless/2350_2351_merged.pdf'
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 327, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/consumer.py", line 698, in run
    self.input_doc.original_file.unlink()
  File "/usr/local/lib/python3.11/pathlib.py", line 1147, in unlink
    os.unlink(self)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/paperless/2350_2351_merged.pdf'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/src/paperless/src/documents/tasks.py", line 151, in consume_file
    msg = plugin.run()
          ^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/consumer.py", line 712, in run
    self._fail(
  File "/usr/src/paperless/src/documents/consumer.py", line 302, in _fail
    raise ConsumerError(f"{self.filename}: {log_message or message}") from exception
documents.consumer.ConsumerError: 2350_2351_merged.pdf: The following error occurred while storing document 2350_2351_merged.pdf after parsing: [Errno 2] No such file or directory: '/tmp/paperless/2350_2351_merged.pdf'
[2024-06-26 10:17:21,155] [ERROR] [celery.app.trace] Task documents.tasks.consume_file[4b7afaee-46f3-4798-b029-767b14828d77] raised unexpected: ConsumerError("2350_2351_merged.pdf: The following error occurred while storing document 2350_2351_merged.pdf after parsing: [Errno 2] No such file or directory: '/tmp/paperless/2350_2351_merged.pdf'")
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 327, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/consumer.py", line 698, in run
    self.input_doc.original_file.unlink()
  File "/usr/local/lib/python3.11/pathlib.py", line 1147, in unlink
    os.unlink(self)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/paperless/2350_2351_merged.pdf'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 453, in trace_task
    R = retval = fun(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 736, in __protected_call__
    return self.run(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 327, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/tasks.py", line 151, in consume_file
    msg = plugin.run()
          ^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/consumer.py", line 712, in run
    self._fail(
  File "/usr/src/paperless/src/documents/consumer.py", line 302, in _fail
    raise ConsumerError(f"{self.filename}: {log_message or message}") from exception
documents.consumer.ConsumerError: 2350_2351_merged.pdf: The following error occurred while storing document 2350_2351_merged.pdf after parsing: [Errno 2] No such file or directory: '/tmp/paperless/2350_2351_merged.pdf'
```


### Browser logs

_No response_

### Paperless-ngx version

2.10.1

### Host OS

Synology DSM 7.2 | x86_64

### Installation method

Docker - official image

### System status

```json
{
    "pngx_version": "2.10.1",
    "server_os": "Linux-4.4.302+-x86_64-with-glibc2.36",
    "install_type": "docker",
    "storage": {
        "total": 1546188226560,
        "available": 822746947584
    },
    "database": {
        "type": "postgresql",
        "url": "paperless",
        "status": "OK",
        "error": null,
        "migration_status": {
            "latest_migration": "paperless_mail.0024_alter_mailrule_name_and_more",
            "unapplied_migrations": []
        }
    },
    "tasks": {
        "redis_url": "redis://paper-redis:6379",
        "redis_status": "OK",
        "redis_error": null,
        "celery_status": "OK",
        "index_status": "OK",
        "index_last_modified": "2024-06-26T10:22:38.990666+02:00",
        "index_error": null,
        "classifier_status": "OK",
        "classifier_last_trained": "2024-06-26T08:05:03.591088Z",
        "classifier_error": null
    }
}
```


### Browser

Edge, Chrome

### Configuration changes

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/documents/bulk_edit.py
+++ b/src/documents/bulk_edit.py
@@ -2,6 +2,7 @@ import hashlib
 import itertools
 import logging
 import os
+import tempfile
 from typing import Optional
 
 from celery import chain
@@ -269,7 +270,7 @@ def merge(
         return "OK"
 
     filepath = os.path.join(
-        settings.SCRATCH_DIR,
+        tempfile.mkdtemp(dir=settings.SCRATCH_DIR),
         f"{'_'.join([str(doc_id) for doc_id in doc_ids])[:100]}_merged.pdf",
     )
     merged_pdf.remove_unreferenced_resources()
@@ -321,7 +322,7 @@ def split(doc_ids: list[int], pages: list[list[int]], delete_originals: bool = F
                 for page in split_doc:
                     dst.pages.append(pdf.pages[page - 1])
                 filepath = os.path.join(
-                    settings.SCRATCH_DIR,
+                    tempfile.mkdtemp(dir=settings.SCRATCH_DIR),
                     f"{doc.id}_{split_doc[0]}-{split_doc[-1]}.pdf",
                 )
                 dst.remove_unreferenced_resources()

